src/components/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import classnames from 'classnames';
+import { getCommonProps } from '../../utils';
+import { Command, CommandInput, CommandList, CommandItem, CommandGroup } from 'cmdk';
+import { Icon } from '../Icon';
+import * as Popover from '@radix-ui/react-popover';
+export interface ComboBoxProps {
+  /**
+   * List of options to be displayed in the ComboBox.
+   */
+  options: {
+    label?: string;
+    value: string;
+  }[];
+  /**
+   * Unique id for the ComboBox.
+   */
+  id: string;
+  /**
+   * Optional className for custom styling.
+   */
+  className?: string;
+  /**
+   * Label for the ComboBox.
+   */
+  label: string;
+  /**
+   * Optional placeholder text for the input.
+   */
+  placeholder?: string;
+  /**
+   * Input value for the ComboBox.
+   */
+  inputValue: string;
+  /**
+   * Passed in function to handle input value changes.
+   */
+  setInputValue: (value: string) => void;
+}
+
+const ComboBox = React.forwardRef<HTMLDivElement, ComboBoxProps>(function ComboBox(
+  { options = [], className, id, label, placeholder, inputValue = '', setInputValue, ...props },
+  ref,
+) {
+  const { className: baseClassName, ...commonProps } = getCommonProps({ id }, 'ComboBox');
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const containerRef = React.useRef<HTMLDivElement | null>(null);
+  const itemRefs = React.useRef<Record<string, HTMLDivElement | null>>({});
+  const sanitizedOptions = options.map((option) => String(option.value).toLowerCase());
+
+  React.useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      const popoverContent = document.querySelector(`.${baseClassName}__content`);
+      if (
+        containerRef.current &&
+        !containerRef.current.contains(event.target as Node) &&
+        popoverContent &&
+        !popoverContent.contains(event.target as Node)
+      ) {
+        setIsOpen(false);
+        if (!sanitizedOptions.some((option) => option === inputValue.toLowerCase())) {
+          setInputValue('');
+        }
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [inputValue, sanitizedOptions, baseClassName, setInputValue]);
+
+  return (
+    <div
+      ref={(node) => {
```

Comment:
ah, are you trying to mergeRefs here?  There's a good utility function to do that `https://floating-ui.com/docs/react-utils`